### ISSUE TITLE
playwright.Browser.launch_persistent_context

### ISSUE THREAD

**Issue by Mattwmaster58** [Mattwmaster58/playwright_stealth#9]
Unsupported via automatic patching. This is because we need a way to get the browsers UA so we can  replace "Headless" in it (and in Sec-CH-UA). With all other methods a temp page is launched, and this is low overhead. But launch_persistent_context would require launching and entire browser.

**Comment by ahgraber** [comment]
Is there a quick-and-dirty hack that would at least let me manually provide the browser UA to launch_persistent_context()?

If so, does it lessen the overhead concerns to either:
1. run a secondary/background process with `launch`, get the UA, and use it for `launch_persistent_context`
2. persist the UA with an env var or post-init update to Stealth() config?

Feel free to tell me I'm an idiot or ignore - I'm not familiar with playwright or JS at all and I'm trying to hack a way to archive webpages -- I'd like to use `launch_persistent_context` with an adblocking extension

**Comment by Mattwmaster58** [comment]
> Is there a quick-and-dirty hack that would at least let me manually provide the browser UA to launch_persistent_context()?

Yes, I should allow this. I just haven't gotten around to making the API for this make sense.

1. providing a UA would eliminate the need to seed one from the browser being launched
2. I would rather not persist anything via any mechanism if I don't have to, I would rather this be offloaded and be made explicit to the users. Implicitly using these values is inviting confusion eg why one machine works and another doesn't with the "same" (explicit) configuration

**Comment by iefsu** [comment]
+1

**Comment by Mattwmaster58** [comment]
I am planning on warning the user when no override is provided for launch persistent context, telling them to do so. If someone wants to open a PR, this would be the place to implement it:

[stealth.py:L438](https://github.com/Mattwmaster58/playwright_stealth/blob/0529a7cba8f110974eafbef08810ee36e63119c5/playwright_stealth/stealth.py#L438-L439)
